spaceship/invite.py

```diff
@@ -0,0 +1,64 @@
+
+from flask import render_template, url_for
+import logging
+import re
+from typing import List, Optional
+
+from spaceship import email, achievements
+from spaceship.models import Invitation, Team, User
+
+log = logging.getLogger('spaceship.invite')
+
+class Invite:
+  DEFAULT_SUBJECT = "Join my Spaceship Earth crew!"
+  EMAIL_REGEX = re.compile(r"""[^@\s,]+@[^\s,]+""")
+
+  @classmethod
+  def default_message(cls, inviter: User, team: Team):
+    return render_template('invite_crew.html', inviter=inviter, team=team)
+
+  @classmethod
+  def perform(cls, inviter: User, team_id: int, subject: str, message: str, emails: List[str]) -> Optional[str]:
+    team = Team.query.get(team_id)
+    if not team or team not in inviter.teams:
+      return f'You are not authorized to invite crew onto team {team_id}'
+
+    subject = subject if subject else cls.DEFAULT_SUBJECT
+    message = message if message else cls.default_message(inviter, team)
+
+    # quilljs needs <p><br></p> to show line breaks even though everything is in a paragraph
+    # this creates extra line breaks in gmail so just strip it out
+    message = message.replace('<p><br></p>', '')
+
+    if not emails:
+      return f'List a few folks to invite!'
+
+    # an email is anything containing an `@` character, separated by whitespace and/or commas
+    for invited_email in cls.EMAIL_REGEX.findall(emails):
+      iv = Invitation(
+        inviter=inviter,
+        team=team,
+        invited_email=invited_email,
+        message=message,
+        status='sent')
+      iv.save()
+
+      # each recipient sees a unique invite link
```

Comment:
is this still true?  seems to contradict your pr message

Reply:
yeah, we still send folks unique invite links which will auto-populate the email address field in the registration. but we ALSO make a non-customized link available.

Reply:
https://i.kym-cdn.com/photos/images/original/000/538/731/0fc.gif